https-proxy-agent.js

```diff
@@ -36,6 +36,9 @@ function HttpsProxyAgent (opts) {
   // if `true`, then connect to the proxy server over TLS. defaults to `false`.
   this.secureProxy = proxy.protocol ? /^https:?$/i.test(proxy.protocol) : false;
 
+  // use custom CA certificate
+  this.ca = opts.ca;
```

Comment:
Hi @TooTallNate,

I don't know enough about the Node APIs to say if `ca` works for `https.request()`, or whether it's only something that can be used with a custom agent option.

The documentation shows the same options going into `https.request()` as `https.Agent()`: https://nodejs.org/api/https.html#https_https_request_options_callback

Problem is, @zkat suggests we'll also need `key`, `certfile`, and `localAddress` which may or may not need a similar fix to this one.

Any suggestions?